common/src/main/java/com/dumbdogdiner/stickyapi/common/util/TextUtil.java

```diff
@@ -113,41 +150,139 @@ public static boolean isCharacterSupported(char c) {
+                width += (component.toPlainText().length() + 1) * 2;
+            }
+        }
+
+        return width;
+    }
+
+    public static String[] splitLines(String... lines) {
+
+        List<String> processedLines = new ArrayList<>();
+        for (String line : lines) {
+            StringBuilder processedLine = new StringBuilder();
+            ChatFormatStack formatStack = new ChatFormatStack();
+            boolean chatCode = false;
+            int width = 0;
+            for (char c : line.toCharArray()) {
+                if (c == ChatColor.COLOR_CHAR) {
+                    chatCode = true;
+                    processedLine.append(c);
+                } else if (chatCode) {
+                    if(formatStack.push(ChatColor.getByChar(c)) == null) {
+                        // Remove format code if it's not valid
+                        processedLine.deleteCharAt(processedLine.length() - 1);
+                    } else {
+                        // Put in the code if it is valid
+                        processedLine.append(c);
+                    }
+                    chatCode = false;
+                } else {
+                    int charWidth = getCharacterWidth(c, formatStack.isBold());
+                    if (width + charWidth > HALF_PIXELS_PER_CHAT_LINE) {
+                        // Start a new line!
+                        processedLines.add(processedLine.toString());
 
-        for (char c : text.toCharArray()) {
-            width += TextUtil.getCharacterWidth(c);
+                        processedLine = new StringBuilder();
+                        width = 0;
+
+                        processedLine.append(formatStack.toString());
+                    }
+                    // Append to the current (possibly new) line
+                    processedLine.append(c);
+                    width += charWidth;
+                }
+            }
+            processedLines.add(processedLine.toString());
         }
 
-        if (isBold)
-            width += 2 *  text.length();
+        return processedLines.toArray(new String[0]);
+    }
 
-        return width;
+    public static String[] prettifyText(String... text) {
+        text = splitLines(text);
+        for (int i = 0; i < text.length; i++) {
+            int width = getStringWidth(text[i]);
+            int pad = Math.round((HALF_PIXELS_PER_CHAT_LINE - width) / 2f / SPACE_WIDTH) / 2;
+            text[i] = Strings.padStart("", pad, ' ') + text[i];
+        }
+        return text;
     }
 
-    /**
-     Measure the width of a string, assuming it is in the default Minecraft font.
-     *
-     * @param text The string to measure
-     * @return The width of the string, in half-pixels
-     * @see #getStringWidth(String, boolean)
-     */
-    public static int getStringWidth(@NonNull String text) {
-        return getStringWidth(text, false);
+
+    @SuppressWarnings("UnusedReturnValue")
+    private static final class ChatFormatStack {
```

Comment:
Internal comment here would be nice, just so us dumb developers can work out what it does.